### ISSUE TITLE
Can't open Dota2 since Dawnbreaker 7.29 patch (Unable to load module server)

### ISSUE THREAD

**Issue by abinderiya** [ValveSoftware/Dota-2#1918]
#### Your system information

* System information from steam (`Steam` -> `Help` -> `System Information`) in a [gist](https://gist.github.com/abinderiya/52523ee5cca21a44e67a1f18f1392ee9): 
* Have you checked for system updates?: [Yes]
* Are you using the latest stable video driver available for your system? [Yes]
* Have you verified the game files?: [Yes]

#### Please describe your issue in as much detail as possible:
I'm unable to launch Dota 2 after the patch update. It is throwing this exception:

CAppSystemDict::LoadSystemAndDependencies(CUtlVect or<CAppSystemDict::AppSystemRequest_t> &, ErrorResponse_t)():
CAppSystemDict:Unable to load module server (Dependency of application), error 2

#### Steps for reproducing this issue:

1. Update to 7.29
2. Launch

**Comment by Turmfalke2** [comment]
This sounds like something verifying the game files should catch, do you have a libserver* or similar file below `steamapps/common/dota 2 beta/game/dota/bin`?

**Comment by abinderiya** [comment]
> This sounds like something verifying the game files should catch, do you have a libserver* or similar file below `steamapps/common/dota 2 beta/game/dota/bin`?

I have verified the game files countless times and it still does not work. I have seen multiple posts from different people on the dev forum pointing to the same problem so it seems like the problem is not with my computer. I am just wondering if it will ever be fixed by valve or dota 2 on macOS is becoming a thing in the past.

**Comment by Turmfalke2** [comment]
Can you link the posts? Might be useful for other people encountering the same issue. Also I doubt this is an issue for all macOS user, we would have far more reports of it here.

Did you find something called libserver?

**Comment by abinderiya** [comment]
> Can you link the posts? Might be useful for other people encountering the same issue. Also I doubt this is an issue for all macOS user, we would have far more reports of it here.
> 
> Did you find something called libserver?

https://dev.dota2.com/forum/bugs/gameplay-bugs/2258688-can-t-start-the-game. This is just one example. I have seen another one on steam community tab. 

I was not able to find libserver. What does it have to do with the problem? Do you work for valve?

**Comment by Turmfalke2** [comment]
I don't, I just try to help people here as much as I can. In the hope the little time valve employees have for linux & mac osx is spend as efficient as possible.   

Back to your issue. I interpret the errors message as a system within the dota 2 client looking for another component (likely in form of an dlopen) and then failing to find it. I suspect the server module is related to running a local dota server as used for the demo mode/local lobby. I don't own a mac so I can only compare it to how my installation here on linux looks like.
In `~/.steam/steam/steamapps/common/dota 2 beta/game/dota/bin/linuxsteamrt64` I have files called 
`libclient.so`,  `libhost.so` and `libserver.so`. My suspicion was that libserver.so might be corrupted/missing.
If I rename/move away my `libserver.so` I get an almost identical error message.   

I don't think mac osx uses .so files. I think they are using .slo? I am unsure here, but I would suspect there to be a similar file. Do you have an antivirus service running that could be interfering with dota locating that file?  

If I remove `libserver.so` I get the following messages: 

![screen02](https://user-images.githubusercontent.com/5174562/114580319-0abefb00-9c7f-11eb-88a9-179a9d58cc12.png)

followed by

![screen03](https://user-images.githubusercontent.com/5174562/114580351-101c4580-9c7f-11eb-8e47-464d6750d926.png)

but then steam triggers file cache verification and redownloads the file.

**Comment by abinderiya** [comment]
> I don't, I just try to help people here as much as I can. In the hope the little time valve employees have for linux & mac osx is spend as efficient as possible.
> 
> Back to your issue. I interpret the errors message as a system within the dota 2 client looking for another component (likely in form of an dlopen) and then failing to find it. I suspect the server module is related to running a local dota server as used for the demo mode/local lobby. I don't own a mac so I can only compare it to how my installation here on linux looks like.
> In `~/.steam/steam/steamapps/common/dota 2 beta/game/dota/bin/linuxsteamrt64` I have files called
> `libclient.so`, `libhost.so` and `libserver.so`. My suspicion was that libserver.so might be corrupted/missing.
> If I rename/move away my `libserver.so` I get an almost identical error message.
> 
> I don't think mac osx uses .so files. I think they are using .slo? I am unsure here, but I would suspect there to be a similar file. Do you have an antivirus service running that could be interfering with dota locating that file?
> 
> If I remove `libserver.so` I get the following messages:
> 
> ![screen02](https://user-images.githubusercontent.com/5174562/114580319-0abefb00-9c7f-11eb-88a9-179a9d58cc12.png)
> 
> followed by
> 
> ![screen03](https://user-images.githubusercontent.com/5174562/114580351-101c4580-9c7f-11eb-8e47-464d6750d926.png)
> 
> but then steam triggers file cache verification and redownloads the file.

Yeah the error is actually identical. So i suspect that steam is somehow failing to locate the missing file on mac because I have verified the game files many times and nothing came out of it. Thanks for your effort.

**Comment by Turmfalke2** [comment]
do you have similar directory with the client/host files?

**Comment by abinderiya** [comment]
> do you have similar directory with the client/host files?

it turns out that i do have the files. And I even tried deleting one of them to see if steam is able to locate it and trigger download. It is downloading just fine.

**Comment by bintang013** [comment]
I thought I'm alone!Me too.

I also have libclient.dylib, libhost.dylib, and libserver.dylib but I still have the error message exactly like @abinderiya 

I tried verify cache countless time, reinstalling dota 2 and even steam three times and nothing happens :(. meanwhile, my CSGO is doing just fine...

Here's my Imac spec:

OS catalina 10.15.7
memory 32gb ddr4
graphic AMD radeon pro 5500 xt 8gb

I also regularly use cleanMyMac

**Comment by Turmfalke2** [comment]
you also have that issue since the dawnbreaker update? Do you have anti virus like software running that may prevent dota from accessing libserver?  

I guess the next step would be to trace dota using dtrace to check whether it finds the file. However since I don't own a mac I don't think I can explain you how to use dtrace.  

Maybe @danginsburg can help you.

**Comment by bintang013** [comment]
Thanks for your response. Damn, I'm not tech savvy, don't understand dtrace. I'll try to search for it first. And for nope, I don't have any antivirus as far as im concern...

**Comment by binjospookie** [comment]
Hi! I faced with the same issue but find how to fix and fixed it.

First of all I run `dota.sh` and what I've got
```
failed to dlopen "/Users/binjo/Library/Application Support/Steam/steamapps/common/dota 2 beta/game/bin/osx64/dota2.app/Contents/MacOS/libserver.dylib" error=dlopen(/Users/binjo/Library/Application Support/Steam/steamapps/common/dota 2 beta/game/bin/osx64/dota2.app/Contents/MacOS/libserver.dylib, 9): image not found
 failed to dlopen "libserver.dylib" error=dlopen(libserver.dylib, 9): image not found
```

Yes. In `dota2.app/Contents/MacOS/libserver.dylib` I had no `libclient.dlyb`, `lib host.dlyb`, `libserver.dlyb`.
But! This files are placed in `/Users/binjo/Library/Application Support/Steam/steamapps/common/dota 2 beta/game/dota/bin/osx64/`

So I copied files from  `/Users/binjo/Library/Application Support/Steam/steamapps/common/dota 2 beta/game/dota/bin/osx64/` to `/Users/binjo/Library/Application Support/Steam/steamapps/common/dota 2 beta/game/bin/osx64/dota2.app/Contents/MacOS/` and run Dota 2 without errors!

Looks like new patch has an imports errors or some files was copied to the wrong folder.

I hope Dota 2 developers will fix it.

<img width="2503" alt="image" src="https://user-images.githubusercontent.com/8722478/115158720-47805d00-a098-11eb-9784-df0c346186ab.png">

**Comment by abinderiya** [comment]
This actually works thanks a lot.

**Comment by bintang013** [comment]
Damn! You are 21st century Messiah indeed @binjospookie ! Super thanks a lot it works!

**Comment by Turmfalke2** [comment]
@binjospookie nice.

Can you try creating symlinks instead of copying the files using `ln -s`? Your copy solution is likely to break with the next update that touches those files.

**Comment by abinderiya** [comment]
@Turmfalke2 Just like you predicted, the new update already broke the fix. To fix it, you need basically need to repeat the steps above and replace the old files.

**Comment by Iliketoplaygames2** [comment]
Hey all, I'm having similar issues I think to all of you folks. I keep getting these error messages... tried a lot of things like redownloading, verifying the game files etc. No luck.
<img width="466" alt="Screen Shot 2021-04-13 at 9 15 23 AM" src="https://user-images.githubusercontent.com/83041604/115792961-35086b00-a399-11eb-8fdb-8e78c2a1c3d1.png">
<img width="545" alt="Screen Shot 2021-04-13 at 9 19 05 AM" src="https://user-images.githubusercontent.com/83041604/115792965-36d22e80-a399-11eb-949f-6367d112f920.png">

**Comment by Turmfalke2** [comment]
That is the same issue, have you tried the steps from https://github.com/ValveSoftware/Dota-2/issues/1918#issuecomment-822049134 ?
